cirq/api/google/v2/device.proto

```diff
@@ -0,0 +1,96 @@
+syntax = "proto3";
+
+package cirq.api.google.v2;
+
+option java_package = "com.google.cirq.api.google.v2";
+option java_outer_classname = "CalibrationProto";
+option java_multiple_files = true;
+
+// This contains information about a device that includes the
+// qubits on the device, supported gates, connections, and timing.
+// This message specifies information that is needed when sending a
+// Program message to the device.
+message DeviceSpecification {
+
+  // A list of allowed gatesets for programs submitted to this processor
+  // Language.gate_set should be one of these values to be valid.
+  repeated GateSet valid_gate_sets = 1;
+
+  // A list of allowed ids for qubits within the Program.
+  // Any programs with ids not in this list will be rejected.
+  // If empty, all qubit values are allowed (e.g. in a simulator)
+  repeated string valid_qubits = 2;
+
+  // A list of allowed targets for multi-qubit gates.
+  // For a typical grid layout, this will be a list of adjacent pairs
+  repeated Target valid_targets = 3;
```

Comment:
I know we would like to keep this small but this might not be flexible enough.  For example we could have two qubit gates, plus measurements that are across one resonator and so couple multiple qubits.  Right now you can't specify that because Target isn't on the gate definition.

Reply:
Hmmm.  What do you think about this idea?

We make the top-level target list the "default" list, but add an option to override the target list at the gate level.  If not specified at the gate-level, it defaults to the top-level list.  That way, for the typical case, it would be the same size, but would allow gate-specific target lists for more complicated gates.

What do you think?

(I have not put this into the proto yet)

Reply:
I further refined this idea after we talked and then decided to make a proto called a TargetList with a list of targets.

The gates then refer to this list by name rather than re-list all targets.  This means that, for the typical case, you just list all the qubit pairs, then give it a name, like "adjacent_pairs" then all the gates say that their valid target is "adjacent_pairs".

What do you think?